### DISCUSSION TITLE
A general login screen that skips the "add server" flow

### DISCUSSION THREAD

**Discussion by EmmaKnijn** [audioling/audioling#13] [Ideas]
I'd love to see a login screen on the web version, where users can simply enter a username and password and log in with a server pre configured in the backend

**Comment by jeffvli** [comment]
This already exists in audioling, though in this case users would need to authenticate twice.

Audioling has its own local server so it runs its own authentication. The flow is as follows:
1. Screen to register / log in to the audioling server
2. Server selection screen with all added Subsonic/Jellyfin servers (by an admin) with individual login for each one

**Comment by EmmaKnijn** [comment]
Okay, so the audioling server isn't an optional component, wow. In my ideal world I'd like to present users with a single login prompt for the entire UI (that preferably also does user management (password update and such)) that can replace the navidrome UI for users.
It's sad that I can't really do development using tauri so I can't help out.

**Reply by jeffvli** [reply]
Right, the server is an integral part of the app since all API requests to your Navidrome server will be routed through it.

The reason for adding a local auth is so that all other users can use your credentials to access your server in a read-only state. It also allows me to eventually add extra functionality that is specific to audioling while still locking it behind authentication. For example, lyrics fetching, group listening, and other features.

The current flow is as follows: 

1. Users will first be presented a login screen (in which they can register the account themselves)
![image](https://github.com/user-attachments/assets/8f493f6a-0284-4aca-86ef-315d374c9872)

2. Then they will be provided a list of libraries (that you have added)
![image](https://github.com/user-attachments/assets/947cd34a-5d90-46e7-b026-d73579f3f235)

3. Once they enter the server selection screen, they have the option to either connect to your server with your credentials (in read-only mode) if you set your server to be a public server, or they can add their own credentials to have write mode which includes playlists, favorites, ratings, etc.

![image](https://github.com/user-attachments/assets/f90d1f23-27ee-4d16-a2ab-8f143c5dbd49)

**Reply by EmmaKnijn** [reply]
That seems pretty good, can I contact you through Discord or something so I talk about my usecase and stuff, I feel like a dick if I just go about sharing what I'm working on in someone else's project

**Reply by jeffvli** [reply]
I'm ok with discussing here since others may also share your use case. Otherwise, my public discord server is linked at the top of the readme.

**Reply by EmmaKnijn** [reply]
Basically I'm building an artist and user focused streaming platform (https://streaming.knijn.one), and I feel like using open source software to do that is an efficient way to go about it.

**Comment by raphmim** [comment]
Thank you @jeffvli for explaining the logic behind the local server, though I'm not sure I'm get it 100% yet 😅. Could you explain what is the use case for having multi-user support in a desktop app? Is it meant for computers that are used by many different people? I'm trying to wrap my head around it and think about how we can simplify the UX, because having to register / log in a local server will be confusing to users who aren't tech savvy (it was confusing to me, and I am tech savvy 😆)

**Reply by jeffvli** [reply]
It's because it's not meant to be a desktop-only app. The intention was to also allow for a web setup for multi-user support with credentials that can bypass the original server (for example if you wanted to host a public server without every user needing an account to authenticate).

I'm starting to realize though that I might want to abandon this idea completely and just transition this to a local-only / offline-first setup without a http server.

**Reply by raphmim** [reply]
As someone who is often traveling and often faces connectivity issues, I'm always supportive of offline-first :) A multi-user web app has a role to play, but perhaps in a second phase?
